### ISSUE TITLE
React "build-storybook" fails in v12.7.0 and Webpack5

### ISSUE THREAD

**Issue by danr-za** [nrwl/nx#6703]
## Current Behavior
React `build-storybook` fails in v12.7.0 & Webpack 5.

## Expected Behavior
Should build without errors

## Steps to Reproduce
* Create a repo with v12.7.0 and React app
* Generate storybook configuration `nx g @nrwl/react:storybook-configuration`
* run `nx run project:build-storybook`

### Failure Logs
```sh
>  NX  Storybook builder starting ...

info => Cleaning outputDir: /Users/ddd/frontend/dist/storybook/foo
info => Loading presets
info => Compiling manager..
info => Compiling preview..
info => Loading 1 config file in "/Users/ddd/frontend/apps/foo/.storybook"
info => Loading 9 other files in "/Users/ddd/frontend/apps/foo/.storybook"
info => Adding stories defined in "/Users/ddd/frontend/apps/foo/.storybook/main.js"
info => Loading Nrwl React Storybook preset from "@nrwl/react/plugins/storybook"

>  NX  Using webpack 5. Reason: detected version 5 in node_modules/webpack/package.json

info => Using default Webpack4 setup
Invalid value used in weak set
TypeError: Invalid value used in weak set
    at WeakSet.add (<anonymous>)
    at MiniCssExtractPlugin.apply (/Users/dan/work/frontend/node_modules/mini-css-extract-plugin/dist/index.js:362:18)
    at webpack (/Users/dan/work/frontend/node_modules/@storybook/builder-webpack4/node_modules/webpack/lib/webpack.js:51:13)
    at Object.build (/Users/dan/work/frontend/node_modules/@storybook/builder-webpack4/dist/cjs/index.js:179:18)
    at async Promise.all (index 1)
    at async buildStaticStandalone (/Users/dan/work/frontend/node_modules/@storybook/core-server/dist/cjs/build-static.js:141:28)

```

### Environment
```sh
  Node : 14.16.1
  OS   : darwin x64
  yarn : 2.4.2
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.7.0
  @nrwl/cypress : 12.7.0
  @nrwl/devkit : 12.7.0
  @nrwl/eslint-plugin-nx : 12.7.0
  @nrwl/express : Not Found
  @nrwl/jest : 12.7.0
  @nrwl/linter : 12.7.0
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/nx-cloud : Not Found
  @nrwl/react : 12.7.0
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.7.0
  @nrwl/web : 12.7.0
  @nrwl/workspace : 12.7.0
  @nrwl/storybook : 12.7.0
  @nrwl/gatsby : Not Found
  typescript : 4.3.5

```

**Comment by juristr** [comment]
Hey, the reason is that the Storybook setup uses Webpack 4 (as can be seen from your Storybook log). Since the default React setup is still Webpack 4, also the Storybook generator uses that as its default when setting up Storybook. We will switch to Webpack 5 also for the Storybook generator once React uses that as the default.

However, you can enable Storybook's Webpack 5 support by adding the "builder" in the `main.js` like

```js
const rootMain = require('../../../.storybook/main');

module.exports = {
  ...rootMain,

  core: {
    builder: 'webpack5',
  },

  ...
};
```

Also, you need to install the following Storybook packages:

```bash
yarn add @storybook/manager-webpack5 @storybook/builder-webpack5
```

(obviously you can also use npm)

**⚠️ Note however,** even by doing the above, there's an issue with the current Nx React Storybook preset which got fixed in #6714 and which will be released with Nx v12.8 soon.

**Comment by kazamov** [comment]
@juristr, the fix for webpack 5 released in 12.7.2, so you can use it

**Comment by danr-za** [comment]
Working as expected in 12.8.0
